### ISSUE TITLE
feat(util): migrating _getPkgPath util function - 50%

### ISSUE THREAD

**Issue by pmvald** [pinglue/pg-repo#39]
This task is along direction of migrating file system util functions to make it possible to "inject" file system module in them to make testing easier.

We want to migrate the _getPkgPath util function at `pinglue/src/registry/utils/helpers.ts:99` to `@pinglue/utils/src/bm-factories/fs-find-pkg-path.ts` and change it signature to:

```
_findPkgPath(pkgName: string, fsModule: FsModule)
```

where fsModule is the file system module which is very similar to fs-extra module (find its more accurate definition by looking at the function `fsFactory`  in `@pinglue/utils/src/bm-factories/fs-factory.ts`)

This task is very similar to what I did for the util function _readYaml (same file as _getPkgPath function). Also write a test file similar to `fs-yaml.spec.ts` where we use an in memory fs module as fsModule argument. You just need to repeat what I did for _readYaml.

What does _getPkgPath(pkgName) function do? It finds the path of the installed package with the given name `pkgname`. Under the hood, it starts from the current directory and walks back to the root, and for each folder in the way it checks whether the path `./node_modules/pkgName` exists. This is the exactly the same algorithm that Node uses for find an installed package ([see here](https://nodejs.org/api/esm.html#resolver-algorithm-specification)).

**Comment by yuki-tylar** [comment]
I'm trying to write spec file for `findPkgPath` but I got a problem.

findUp is using package `fs` inside. (`locate-path` package actually) 
So, even if I create `fakeFs` with some data and `matcher` in `findUp` returns some strings (which is pass), `findUp` checks real `fs` after the `matcher` process, so it always return `undefined`.

To solve this, maybe we just need to use `fakeFs.pathExists` to check if the path exist instead of using `findUp`.

how do you think?

**Comment by pmvald** [comment]
of course! the whole idea of writing all these util modules is to solve this problem of hooking to the real fs. All these libraries hard-code the real fs and this makes it difficult for us to test our libraries. You need to drop findup lib. Right now we are blocked from further progress because of this problem that we cannot write tests for the our new features. So we need to build a bunch of libraries where fs (and other boundary modules like http) is "injectable" rather than hard-coded. Only after writing these functions we can start making more progress. This has always been a problem and now it is even a bigger problem due to ES6 modules. So time for new libraries. If they wrk well we will publish them as standalone packages later.
